api/src/test/java/org/openmrs/ConditionTest.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.junit.jupiter.api.Test;
+import org.openmrs.api.APIException;
+
+import java.lang.reflect.Field;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * This class tests all methods that are not getter or setters in the Condition java object TODO: finish
+ * this test class for Condition
+ * 
+ * @see Condition
+ */
+public class ConditionTest {
```

Comment:
Rather than all this I'd add a couple of test cases in [`ConditionServiceImplTest`](https://github.com/icrc-loliveira/openmrs-core/blob/a83975187495d8e438dc2cc74b4da2fc00838adb/api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java#L37).

Reply:
Hi @mks-d,

I don't think that is the best approach. Since for that I will have to make changes in `ConditionServiceImplTest` to a test a code that is in another class.

Point can be made that I am not testing the correct class. So I made some search how o test an abstract class and find [this](https://www.baeldung.com/junit-test-abstract-class) and decide to try the `3.1.` approach. Let me know your thoughts.

Reply:
@icrc-loliveira `ConditionServiceImplTest` would validate that all is ok with Spring + Hibernate. But the Liquibase changeset can only be tested at app runtime.

Reply:
I updated the test `saveCondition_shouldSaveNewCondition()` in `ConditionServiceImplTest`  to test the `formNamespaceAndPath`field.